smartsim/_core/mli/infrastructure/worker/worker.py

```diff
@@ -507,49 +544,44 @@ def fetch_inputs(
         :raises ValueError: If neither an input key or an input tensor are provided
         :raises SmartSimError: If a tensor for a given key cannot be retrieved
         """
-        fetch_results = []
-        for request in batch.requests:
-            if request.raw_inputs:
-                fetch_results.append(
-                    FetchInputResult(request.raw_inputs, request.input_meta)
-                )
-                continue
+        if not batch.raw_inputs and not batch.input_keys:
```

Comment:
There are a couple issues with this refactor.
- this condition and the one at line 550 will always evaluate to `True`, because `batch.raw_inputs` and `batch.input_keys` are lists (possibly containing `None` many times) and are never empty. This would mean that even if `raw_inputs==[None]`, we would assign that to `FetchInputResults` and return at line 551.
- there is no guarantee that a user is not mixing `raw_inputs` and `input_keys` in the same (batched) request. Thus we need to iterate over `zip(raw_inputs, input_keys)` and resolve each couple individually, storing the resulting `raw_inputs` and `input_meta` as lists that are assigned then passed to `FetchInputResult` once the iterations are over.

Reply:
Here's how I was understanding it but I could be totally wrong. In `InferenceRequest`, passed in None values for `raw_inputs,` `input_keys`, `input_meta`, and `output_keys` have `self.raw_inputs = raw_inputs or []`, so if  a `None` value is passed in, it becomes an empty list. When we instantiate a `RequestBatch` using `from_requests`, I do it like this:
```python
raw_inputs=[
       request.raw_inputs for request in requests if request.raw_inputs
],
```
meaning that if all the requests have an empty list in place of `raw_requests`, for example, they will just get skipped over and those empty lists are not placed into the `batch.raw_requests` list, leaving `batch.raw_requests` an empty list.

I added this test just to be sure.
```python
def test_fetch_inputs_no_input_source():
    """Verify that the ML worker fails when there are no inputs provided"""
    worker = MachineLearningWorkerCore
    request1 = InferenceRequest()
    request2 = InferenceRequest()

    batch = RequestBatch.from_requests([request1, request2], ModelKey("test-model", "desc"))

    with pytest.raises(ValueError) as exc:
        fetch_result = worker.fetch_inputs(batch, {})
    assert str(exc.value) == "No input source"
```

Reply:
But in regards to mixing `input_keys` and `raw_inputs`, I definitely didn't think about that! I will fix that asap

Reply:
I see your point, you're right, an empty list is possible and that would be a False-y value.

Reply:
I believe I've resolved both of your points in the original comment!